### PR TITLE
:electron: Packaging different architectures and installers for windows

### PR DESCRIPTION
**Pull request by MikesGlitch**
The current windows release is built only for x64 (the architecture the build machine is using). Nobody's complained yet, but if someone comes along with an Arm CPU it wont work.

With this change we're publishing NSIS and APPX (windows store) installers for Windows, in the following architectures
- x64
- ia32
- arm64

**The NSIS files:** 

These files should be kept - they allow users to install without going to the Microsoft Appstore.

**The AppX files:** 

The AppX files are used for the Microsoft Appstore. We've chosen the AppX over the .exe because the AppX installers are recommended, and allow us to host the files on the Windows hosting service. 

**NOTE:** There's a lot of images in here but I think it's necessary. It's to cover all of Windows's scenarios - taskbar icons, start menu, different screen sizes, light/dark mode etc. We're forced to use at minimum 4/5 icons, but it looks terrible. 

**In the future: ** we will automate the release using GitHub actions. For now it's manual because there are tweaks being made to the store listing and every change takes 3(ish) days for Microsoft to approve.

### CODE REVIEW

**Comment by Jonathan-Fang** [comment]
Writing this to give my support for this feature, I think packaging ARM is good future-proofing the app. I personally don't use x32 but it can't hurt right?

File: bin/package-electron [hunk 1]
```diff
@@ -50,10 +50,10 @@ yarn workspace desktop-electron update-client
         if [ -f ../../.secret-tokens ]; then
             source ../../.secret-tokens
         fi
-        yarn build --publish never --arm64 --x64
+        yarn build
```
Comment (MikesGlitch) [inline bin/package-electron:53]: Removing the overrides to the build params, the build script sets it all correctly

File: packages/desktop-electron/package.json [hunk 2]
```diff
@@ -3,13 +3,13 @@
   "author": "Actual",
   "productName": "Actual",
   "description": "A simple and powerful personal finance system",
-  "version": "24.8.0",
+  "version": "24.8.1",
```
Comment (MikesGlitch) [inline packages/desktop-electron/package.json:6]: Every time we make changes to the store listing we need to increment - this means we may go out of sync with the main package

File: packages/desktop-electron/package.json [hunk 3]
```diff
@@ -24,12 +24,6 @@
       "!node_modules/better-sqlite3/{benchmark,src,bin,docs,deps,build/Release/obj,build/Release/sqlite3.a,build/Release/test_extension.node}",
       "build"
     ],
-    "publish": {
```
Comment (MikesGlitch) [inline packages/desktop-electron/package.json:27]: This is old - we used to publish to github for autoupdate but that hasn't worked in ages.

File: packages/desktop-electron/package.json [hunk 4]
```diff
@@ -41,7 +35,7 @@
       },
       "target": [
         {
-          "target": "default",
+          "target": "dmg",
```
Comment (MikesGlitch) [inline packages/desktop-electron/package.json:38]: being specific about what we want from the mac build

File: packages/desktop-electron/package.json [hunk 5]
```diff
@@ -3,13 +3,13 @@
   "author": "Actual",
   "productName": "Actual",
   "description": "A simple and powerful personal finance system",
-  "version": "24.8.0",
+  "version": "24.8.3",
```
Comment (MikesGlitch) [inline packages/desktop-electron/package.json:6]: Every time we make changes to the store listing we need to increment - this means we may go out of sync with the main package

**Comment by youngcw** [comment]
does the appx file only work when installing from the app store? or can that be installed locally?

**Comment by youngcw** [comment]
Does the windows store really require 80 different icon sizes?

**Comment by MikesGlitch** [comment]
> Does the windows store really require 80 different icon sizes?

I agree this is extremely annoying. I think technically you only need 4 images at a minimum. When I did this, the images were either too small or too large in some areas. 

e.g. the taskbar image was too small, the window switcher, the start menu icon in list view/square view/search view. The icons have different backing plates whether it's in light mode or dark mode etc. These images are used in the store too, so it's the image in the listing view/the store search etc.

It's frustrating because it's always the same image, just different sizes. 

The real issue is I can delete some of them, but I'd have no idea what it would affect.

**Comment by MikesGlitch** [comment]
> does the appx file only work when installing from the app store? or can that be installed locally?

I haven't been able to install it without the store. The issue is to do with code signing - we'd need to buy a cert if we wanted to install it - the Windows store gives that to us free of charge. 

I tried with a self signed one cert and couldn't get it to work - there's issues on electron-builder about it, seems loads of people have this issue. 😢

**Comment by youngcw** [comment]
> > Does the windows store really require 80 different icon sizes?
> 
> I agree this is extremely annoying. I think technically you only need 4 images at a minimum. When I did this, the images were either too small or too large in some areas.
> 
> e.g. the taskbar image was too small, the window switcher, the start menu icon in list view/square view/search view. The icons have different backing plates whether it's in light mode or dark mode etc. These images are used in the store too, so it's the image in the listing view/the store search etc.
> 
> It's frustrating because it's always the same image, just different sizes.
> 
> The real issue is I can delete some of them, but I'd have no idea what it would affect.

Does is support svg?  I don't know if we have an svg of the desktop app icon.  Do we?

**Comment by MikesGlitch** [comment]
> Does is support svg? I don't know if we have an svg of the desktop app icon. Do we?

I don't believe so, if you upload the package without these 4, it rejects the package:

Logo: StoreLogo.png
Square150x150Logo: Square150x150Logo.png
Square44x44Logo: Square44x44Logo.png
Wide310x150Logo: Wide310x150Logo.png

It was all generated using the Microsoft Asset Generation tool, it's what they recommend for store apps: https://learn.microsoft.com/en-us/windows/apps/design/style/iconography/visual-studio-asset-generation

**Comment by youngcw** [comment]
> > Does is support svg? I don't know if we have an svg of the desktop app icon. Do we?
> 
> I don't believe so, if you upload the package without these 4, it rejects the package:
> 
> Logo: StoreLogo.png Square150x150Logo: Square150x150Logo.png Square44x44Logo: Square44x44Logo.png Wide310x150Logo: Wide310x150Logo.png
> 
> It was all generated using the Microsoft Asset Generation tool, it's what they recommend for store apps: https://learn.microsoft.com/en-us/windows/apps/design/style/iconography/visual-studio-asset-generation

Its not a big deal if thats what we need.

**Review by youngcw — APPROVED** [review]

**Pull request merged**